File: src/datastructure/queue/MyQueue.java
package datastructure.queue;

import java.util.ArrayList;
import java.util.LinkedList;

public class MyQueue <T>{
    ArrayList<T>  queue = new ArrayList<T>();

    public T dequeue(){
        if(!queue.isEmpty()){
            return queue.remove(0);
        }
        return null;
    }

    public boolean enqueue(T item){
        return queue.add(item);
    }

    public boolean isEmpty(){
        return queue.isEmpty();
    }

    public static void main(String[] args) {
        MyQueue<Integer> mq = new MyQueue<Integer>();
        mq.enqueue(1);
        mq.enqueue(2);
        mq.enqueue(3);
        System.out.println(mq.dequeue());
        System.out.println(mq.dequeue());
        System.out.println(mq.dequeue());
    }
}

File: src/datastructure/queue/MyQueue.kt
package datastructure.queue

class MyQueue<T>(
    private val mutableList: MutableList<T> = mutableListOf()
) {
    fun enqueue(item: T): Boolean {
        return mutableList.add(item)
    } fun dequeue(): T? {
        return if (mutableList.isNotEmpty()) mutableList.removeAt(0) else null
    }
    fun isEmpty(): Boolean = mutableList.isEmpty()
}

fun main() {
    val q: MyQueue<String> = MyQueue()
    q.enqueue("Hello1")
    q.enqueue("Hello2")
    q.enqueue("Hello3")
    println(q.dequeue())
    println(q.dequeue())
    println(q.dequeue())
}
